### PR TITLE
[docs] Add docs for migrating from expo webpack to expo router

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Resolve ENG-9820
- Resolve ENG-9037
- Resolve ENG-8667

# How

Add a bunch of public resources for helping users understand the relationship between `@expo/webpack-config` and Expo Router.

### CODE REVIEW

File: docs/pages/guides/customizing-webpack.mdx [hunk 1]
```diff
@@ -45,39 +45,6 @@ module.exports = async function (env, argv) {
 };
 ```
 
-## Polyfills
```
Comment (EvanBacon) [inline docs/pages/guides/customizing-webpack.mdx:48]: ResizeObserver has 95% global browser usage, so I dropped this section. It's not strictly related to the entire PR.

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+Expo Webpack didn't have built-in service worker support, but you could add it yourself by using the `workbox-webpack-plugin` and adding it to the `webpack.config.js`.
+
+Workbox doesn't have a Metro integration, but because Workbox doesn't require one of the core features of a bundler (transformation, resolution, serialization), it can easily be used as a post-build step.
+
+Simply follow the [guide for using Workbox CLI](https://developer.chrome.com/docs/workbox/modules/workbox-cli/), and wherever it refers to a "build script" use `npx expo export -p web`.
+
+For example, here's a possible flow for setting up Workbox:
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:168]:
```suggestion
For example, here's a possible flow for setting up Workbox. Create a new project with the following command:
```

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+
+- `bun create expo -t tabs my-app`
+- `cd my-app`
+
+Next, create a root HTML file for the app and add the service worker registration script:
+
+```js app/+html.tsx
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:175]:
````suggestion
```tsx app/+html.tsx
````

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    '$ ? Which file types would you like to precache? js, html, ttf, ico, json',
+    '$ ? Where would you like your service worker file to be saved? dist/sw.js',
+    '$ ? Where would you like to save these configuration options? workbox-config.js',
+    "$ ? Does your web app manifest include search parameter(s) in the 'start_url', other than 'utm_' or 'fbclid' (like '?source=pwa')? No",
+  ]}
+/>
+
+Finally, run `workbox generateSW workbox-config.js` to generate the service worker config.
+
+Going forward, you can add a build script to run both scripts in the correct order:
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:248]:
```suggestion
Finally, run `workbox generateSW workbox-config.js` to generate the service worker config. Going forward, you can add a build script in **package.json** to run both scripts in the correct order:
```

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+{
+  "scripts": {
+    "build:web": "expo export -p web && workbox generateSW workbox-config.js"
+  }
+}
+```
+
+## PWA manifests
+
+Unlike `@expo/webpack-config`, Expo Router does not automatically attempt to generate the PWA manifest configuration. You can create one in your `public/manifest.json`:
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:260]:
```suggestion
Unlike `@expo/webpack-config`, Expo Router does not automatically attempt to generate the PWA manifest configuration. You can create one in **public/manifest.json**:
```

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+  "start_url": ".",
+  "display": "standalone",
+  "theme_color": "#000000",
+  "background_color": "#ffffff"
+}
+```
+
+You can link this in your HTML file using the `link` tag:
+
+```js app/+html.tsx
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:292]:
````suggestion
```tsx app/+html.tsx
````

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+      </head>
+      <body>{children}</body>
+    </html>
+  );
+}
+```
+
+## Bundler plugins
+
+If you were using custom bundler plugins, refer to the [Expo Metro config](/versions/latest/config/metro) to learn more about adding custom functionality to your bundler pipeline.
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:335]:
```suggestion
If you were using custom bundler plugins, see [Expo Metro config](/versions/latest/config/metro) for adding custom functionality to your bundler pipeline.
```

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+
+## Navigation
+
+If you used React Navigation for navigating between screens in `@expo/webpack-config`, then refer to the migration guide for [React Navigation](/router/migrate/from-react-navigation).
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:339]:
```suggestion
If you used React Navigation for navigating between screens in `@expo/webpack-config`, see the [migration guide for React Navigation](/router/migrate/from-react-navigation).
```

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+
+## Deployment
+
+Visit the [publishing websites documentation](/distribution/publishing-websites) and select "Expo Router" to learn how to deploy Expo Router websites to various hosting providers.
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:343]:
```suggestion
See [publishing websites](/distribution/publishing-websites) and select "Expo Router" on how to deploy Expo Router websites to various hosting providers.
```

**Review by amandeepmittal — APPROVED** [review]
Lgtm! Thank you for incorporating suggestions 🙏

File: docs/pages/router/migrate/from-expo-webpack.mdx [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+
+Expo Webpack didn't have built-in service worker support. However, you could add it yourself by using the `workbox-webpack-plugin` and adding it to the **webpack.config.js**.
+
+Workbox doesn't have a Metro integration, but because Workbox doesn't require one of the core features of a bundler (transformation, resolution, serialization), it can easily be used as a post-build step. Follow the guide for [using Workbox CLI](https://developer.chrome.com/docs/workbox/modules/workbox-cli/), and wherever it refers to a "build script" use `npx expo export -p web` instead.
+
+For example, here's a possible flow for setting up Workbox. Create a new project with the following command:
+
+<Terminal cmd={
```
Comment (amandeepmittal) [inline docs/pages/router/migrate/from-expo-webpack.mdx:166]:
```suggestion
<Terminal cmd={[
```

This was missing from the changes I suggested earlier and will help fix the CI issue.

**Pull request merged**